### ISSUE TITLE
Method not found hook=shell/loadLibrary (java.lang.Runtime)

### ISSUE THREAD

**Issue by FGKD** [M66B/XPrivacy#2424]
On starting XPrivacy, I am getting an internal check error:

> XPrivacy: 3.6.19
Android: 7.1.2 (SDK 25)
>
> Brand: oneplus
Manufacturer: OnePlus
Model: A0001
Product: bacon
Device: A0001
Host: sultan-box
Display: NJH47F release-keys
Id: NJH47F
>
> Method not found hook=shell/loadLibrary (java.lang.Runtime)

ROM is by sultanxda: Unofficial LineageOS 14.1 with 3.4.0 kernel [Aug. 31, 2017]
XPosed used is unofficial: 26-07-17-xposed-v87-sdk25-Xposed-by-PurifyOS

**Comment by ndorf** [comment]
Same error with HTC's Android 7.0 ROM, official Xposed.

I haven't seen any loadLibrary ondemand prompts, so I'm assuming that function isn't being blocked. Most others appear to work, though. I do get a second internal check error, along with the same one in this issue: Incompatible java.net.InterfaceAddress (followed by class definition). I don't know what exactly that's affecting, though: network/internet blocking appears to work.

**Comment by FGKD** [comment]
Issue still persists with
ROM: [ROM+KERNEL][7.1.2] Unofficial LineageOS 14.1 with 3.4.0 kernel [Oct. 22, 2017]
XPosed: [OFFICIAL] Xposed for Lollipop/Marshmallow/Nougat [v88.1, 2017/10/10]

**Comment by ndorf** [comment]
I am interested in fixing these, but don't (yet) know enough about Android and its APIs to do so easily. I intend to take a stab at it, but don't expect anything too soon.

**Comment by Unpublished** [comment]
https://github.com/Unpublished/XPrivacy/commit/f947637d6ea6ca974371e6b953914070208e4174 should fix this, load0 not testet yet.

**Comment by Cerberus-tm** [comment]
@Unpublished Wow, it's great that you're fixing things! Although I am currently unable to test it myself.

**Comment by M66B** [comment]
@Unpublished I have checked your changes. Maybe it is too early to say this, but simply disabling things is not a real solution. Also, changing things in such a way that it is only compatible with Android N is not a good solution.

To others: this still applies: https://forum.xda-developers.com/showpost.php?p=74200941&postcount=18269

**Comment by Unpublished** [comment]
@M66B Yeah I totally agree with you, but couldn't think of a easy way to change the hook method and use the same setting but it should be a simple if statement in the xruntime constructor or did you already take another approach (couldn't find one yesterday in a quick search).

The disabled hooks caused crashes due to deadlocks (webview) and spammy logs that will require more time to debug and fix.

**Comment by M66B** [comment]
@Unpublished I am not working on XPrivacy anymore, so I am not taking "_another approach_". Also, a real developer can find his/her own solutions. Be aware that if you are going to publish your work, which you more or less did by referencing it in this issue, that you are also taking responsibility for the privacy of others.
